### PR TITLE
Try removal of images first based on ID, then followed by repo tags

### PR DESCRIPTION
**Pull request by xbglowx**
Don't care about being nice when it comes to removal of images.
We will try all expired images based on the operator's -ia input.
If there are still dependencies, the removal will fail, which is
OK.

### CODE REVIEW

**Comment by xbglowx** [comment]
@toli @docker-infra/infra

**Comment by xbglowx** [comment]
Forgot to mention that I also did some golint clean up.

File: docker-proper.go [hunk 1]
```diff
@@ -140,22 +147,24 @@ func getExpired(client *dockerclient.DockerClient) (expiredContainers []*dockerc
...
-	for _, image := range images {
+	for _, image := range expiredImages {
 		log.Print("rmi ", image.Id)
 		if !*dry {
 			if _, err := client.RemoveImage(image.Id, false); err != nil {
-				debug("Couldn't remove image: %s", err)
+				debug("Couldn't remove image with ID %s. Let's try to remove the same image, but this time using repo tags: %s", image.Id, err)
+			}
+		}
+		for _, repoTag := range image.RepoTags {
```
Comment (toli) [inline docker-proper.go]: if it successfully removed, do you still need this loop? 
will you try to remove something that's gone? and i guess that'll fail too, which is fine...

**Comment by toli** [comment]
not sure you need to try to optimize and not remove things that successfully got removed once, that's probably ok

LGTM

File: docker-proper.go (hunk 1 above)
Comment (xbglowx) [inline docker-proper.go]: Good, point. I believe I had this under the error if and then later moved it out of that scope to make the log.Print work better. Let me revisit this.

**Comment by xbglowx** [comment]
@toli PR updated.

**Comment by toli** [comment]
cool, LGTM

**Pull request merged**